src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -98,12 +110,20 @@ public static Builder builder() {
         private volatile HttpServerMechanismsResponder successResponder;
 
         private boolean authenticate() throws HttpAuthenticationException {
+            return evaluate((ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException>) mechanism -> mechanism.evaluateRequest(this));
+        }
+
+        private boolean logout() throws HttpAuthenticationException {
+            return evaluate((ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException>) mechanism -> mechanism.evaluateLogoutRequest(this));
+        }
+
+        private boolean evaluate(ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException> mechanism) throws HttpAuthenticationException {
             List<HttpServerAuthenticationMechanism> authenticationMechanisms = mechanismSupplier.get();
             responders = new ArrayList<>(authenticationMechanisms.size());
             try {
                 for (HttpServerAuthenticationMechanism nextMechanism : authenticationMechanisms) {
                     currentMechanism = nextMechanism;
```

Comment:
Can't help thinking logout should be more of a special case.

Trying to insert it into this loop we now have some mechanisms that could register to send a response to the client, the remaining methods that don't implement the method don't get their chance to send a response to the client.